### PR TITLE
Semantic search

### PR DESCRIPTION
**Pull request by rsdy**

### CODE REVIEW

File: .github/workflows/client-test.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: Client Tests
 on:
   pull_request:
     types: [opened, synchronize]
-    branches: [main]
+    branches: [main, semantic]
```
Comment (nerdypepper) [inline .github/workflows/client-test.yml:6]: we can revert this change with the merge of `semantic`. cc @anastasiya1155.

File: .github/workflows/server-test.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ name: Server Unit Tests
 
 on:
   pull_request:
-    branches: [main]
+    branches: [main, semantic]
```
Comment (nerdypepper) [inline .github/workflows/server-test.yml:5]: likewise, this hunk can be reverted.

File: .github/workflows/tauri-test.yml [hunk 3]
```diff
@@ -3,7 +3,7 @@ name: Tauri Tests
 on:
   pull_request:
     types: [opened, synchronize]
-    branches: [main, dev]
+    branches: [main, semantic]
```
Comment (nerdypepper) [inline .github/workflows/tauri-test.yml:6]: revert this hunk on merge.

File: apps/desktop/src-tauri/src/qdrant.rs [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+    path::Path,
+    process::{Child, Command},
+    time::Duration,
+};
+
+use tauri::{plugin::Plugin, Runtime};
+
+use super::relative_command_path;
+
+pub(super) const QDRANT_CONFIG: &str = r#"
```
Comment (nerdypepper) [inline apps/desktop/src-tauri/src/qdrant.rs:12]: store this as a yaml file elsewhere and use `include_str!` instead.

File: apps/desktop/src-tauri/src/qdrant.rs [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+                .replace(
+                    "{{ STORAGE }}",
+                    &qdrant_dir.join("storage").to_string_lossy(),
+                )
+                .replace(
+                    "{{ SNAPSHOTS }}",
+                    &qdrant_dir.join("snapshots").to_string_lossy(),
+                ),
+        )
+        .unwrap();
```
Comment (nerdypepper) [inline apps/desktop/src-tauri/src/qdrant.rs:177]: there seems to be some trivial templating going on here, we can avoid using string replacements here by doing two things:

- put the template config into `config.yaml` and use rust's format string syntax, i.e., use `{storage}` for template holes
- achieve templating with the `format!` and `include_str!` macros, `format!(inculude_str!("./config.yaml"), storage = storage, snapshots = snapshots)`

File: server/bleep/src/ctags.rs [hunk 6]
```diff
@@ -51,7 +51,7 @@ fn find_files(path: &Path) -> Vec<String> {
         })
         .filter(|de| matches!(de.file_type(), Some(ft) if ft.is_file()))
         .map(|de| {
-            dunce::canonicalize(de.path())
+            crate::canonicalize(de.path())
```
Comment (nerdypepper) [inline server/bleep/src/ctags.rs:54]: @llogiq i am not sure if there is much to be gained here by conditionally compiling dunce as a dependency. internally, it uses a cfg flag to point to `fs::canonicalize` on non-windows targets (IIRC), and other targets should end up with zero extra compilation. for what its worth, dunce is a 100 line library, we might as well vendor it in if required.

File: server/bleep/src/intelligence/language.rs [hunk 7]
```diff
@@ -55,6 +55,9 @@ pub struct TSLanguageConfig {
     /// Compiled tree-sitter scope query for this language.
     pub scope_query: MemoizedQuery,
 
+    /// Compiled tree-sitter chunk query for this language.
+    pub chunk_query: Option<MemoizedQuery>,
+
```
Comment (nerdypepper) [inline server/bleep/src/intelligence/language.rs:60]: this can be removed.

**Review by nerdypepper — CHANGES_REQUESTED** [review]

File: server/bleep/src/ctags.rs (hunk 6 above)
Comment (llogiq) [reply server/bleep/src/ctags.rs:54]: I know it's a small crate, but adding it unconditionally means we download and compile it everywhere. Yes, even macros are compiled. So even if it's just a second to download it and another half to compile, we end up winning one and a half second per non-windows build. A small win, yes, but those tend to add up.

Comment (nerdypepper) [reply server/bleep/src/ctags.rs:54]: makes sense.

**Review by anastasiya1155 — DISMISSED** [review]

File: apps/desktop/src-tauri/src/qdrant.rs (hunk 5 above)
Comment (nerdypepper) [reply apps/desktop/src-tauri/src/qdrant.rs:177]: https://github.com/BloopAI/bloop/pull/96

**Review by ipauler — APPROVED** [review]
